<?xml version="1.0"?>
<robot name="tez_cell" version="1.0">

  <link name="base_link">
    <visual>
      <origin rpy="0 0 0" xyz="-3.414199 -1.4075 -0.325"/>
      <geometry>
        <box size="14.546 11.885 0.65"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-3.414199 -1.4075 -0.325"/>
      <geometry>
        <box size="14.546 11.885 0.65"/>
      </geometry>
    </collision>
  </link>

  <!-- Jig -->
  <joint name="jig_pos" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="base_link"/>
    <child link="jig"/>
  </joint>

  <link name="jig">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://tez_description/mesh/reduced_details/jig.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://tez_description/mesh/reduced_details/jig.dae"/>
      </geometry>
    </collision>
  </link>

  <!-- Linear Axis -->
  <joint name="linear_axis_pos" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="base_link"/>
    <child link="linear_axis"/>
  </joint>

  <link name="linear_axis">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://tez_description/mesh/full_details/linear_axis.dae"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://tez_description/mesh/full_details/linear_axis.dae"/>
      </geometry>
    </collision>
  </link>

  <!-- KR210_1 -->
  <joint name="kr210_1_linear_axis_joint" type="prismatic">
    <origin rpy="0 0 3.14159" xyz="0 0 0.487"/>
    <parent link="linear_axis"/>
    <child link="KR210_1_socket"/>
    <axis xyz="1.0 0.0 0.0"/>
    <limit effort="100" lower="0" upper="9" velocity="100"/>
  </joint>

  <link name="KR210_1_socket">
    <visual>
      <origin rpy="0 0 0" xyz="-0.059841 0.000000 0.325856"/>
      <geometry>
        <box size="1.139659 1.019947 0.651711"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.059841 0.000000 0.325856"/>
      <geometry>
        <box size="1.139659 1.019947 0.651711"/>
      </geometry>
    </collision>
  </link>

  <joint name="kr210_1_joint_1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.5256"/>
    <parent link="KR210_1_socket"/>
    <child link="KR210_1_A1"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit effort="100" lower="-3.22886" upper="3.22886" velocity="100"/>
  </joint>

  <link name="KR210_1_A1">
    <visual>
      <origin rpy="0 0 0" xyz="-0.014046 -0.009306 0.336047"/>
      <geometry>
        <box size="0.631088 1.113008 0.692095"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.014046 -0.009306 0.336047"/>
      <geometry>
        <box size="0.631088 1.113008 0.692095"/>
      </geometry>
    </collision>
  </link>

  <joint name="kr210_1_joint_2" type="revolute">
    <origin rpy="0 0 0" xyz="0.005 -0.35 0.45"/>
    <parent link="KR210_1_A1"/>
    <child link="KR210_1_A2"/>
    <axis xyz="1.0 0.0 0.0"/>
    <limit effort="10000" lower="-2.0944" upper="1.22173" velocity="100"/>
  </joint>

  <link name="KR210_1_A2">
    <visual>
      <origin rpy="0 0 0" xyz="-0.219836 -0.002396 0.656578"/>
      <geometry>
        <box size="0.449671 0.532510 1.714843"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.219836 -0.002396 0.656578"/>
      <geometry>
        <box size="0.449671 0.532510 1.714843"/>
      </geometry>
    </collision>
  </link>

  <joint name="kr210_1_joint_3" type="revolute">
    <origin rpy="0 0 0" xyz="-0.183 0 1.35"/>
    <parent link="KR210_1_A2"/>
    <child link="KR210_1_A3"/>
    <axis xyz="1.0 0.0 0.0"/>
    <limit effort="100" lower="-2.0944" upper="2.70526" velocity="100"/>
  </joint>

  <link name="KR210_1_A3">
    <visual>
      <origin rpy="0 0 0" xyz="0.263750 -0.262250 -0.034810"/>
      <geometry>
        <box size="0.527500 1.399500 0.429352"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.263750 -0.262250 -0.034810"/>
      <geometry>
        <box size="0.527500 1.399500 0.429352"/>
      </geometry>
    </collision>
  </link>

  <joint name="kr210_1_joint_4" type="revolute">
    <origin rpy="0 0 0" xyz="0.178 -0.962 -0.041"/>
    <parent link="KR210_1_A3"/>
    <child link="KR210_1_A4"/>
    <axis xyz="0.0 1.0 0.0"/>
    <limit effort="100" lower="-6.10865" upper="6.10865" velocity="100"/>
  </joint>

  <link name="KR210_1_A4">
    <visual>
      <origin rpy="0 0 0" xyz="0.000000 -0.258565 -0.000750"/>
      <geometry>
        <box size="0.246000 0.537131 0.247500"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.000000 -0.258565 -0.000750"/>
      <geometry>
        <box size="0.246000 0.537131 0.247500"/>
      </geometry>
    </collision>
  </link>

  <joint name="kr210_1_joint_5" type="revolute">
    <origin rpy="0 0 0" xyz="0 -0.438 0"/>
    <parent link="KR210_1_A4"/>
    <child link="KR210_1_A5"/>
    <axis xyz="1.0 0.0 0.0"/>
    <limit effort="100" lower="-2.18166" upper="2.18166" velocity="100"/>
  </joint>

  <link name="KR210_1_A5">
    <visual>
      <origin rpy="0 0 0" xyz="0.023766 -0.050459 0.000000"/>
      <geometry>
        <box size="0.324961 0.341083 0.240222"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.023766 -0.050459 0.000000"/>
      <geometry>
        <box size="0.324961 0.341083 0.240222"/>
      </geometry>
    </collision>
  </link>

  <joint name="kr210_1_joint_6" type="revolute">
    <origin rpy="0 0 0" xyz="0 -0.24 0"/>
    <parent link="KR210_1_A5"/>
    <child link="KR210_1_A6"/>
    <axis xyz="0.0 1.0 0.0"/>
    <limit effort="100" lower="-6.10865" upper="6.10865" velocity="2"/>
  </joint>

  <link name="KR210_1_A6">
    <visual>
      <origin rpy="0 0 0" xyz="0.000000 0.017500 0.000000"/>
      <geometry>
        <box size="0.2 0.035 0.2"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.000000 0.017500 0.000000"/>
      <geometry>
        <box size="0.2 0.035 0.2"/>
      </geometry>
    </collision>
  </link>

  <!-- <link name="EndEffector_Link" />

  <joint name="EndEffector" type="fixed">
    <origin xyz="0 0 -0.061525" rpy="0 0 0" />
    <parent link="KR210_1_A6" />
    <child link="EndEffector_Link" />
    <axis xyz="0 0 0" />
  </joint> -->


  <!-- KR210_2 -->
  <joint name="kr210_2_linear_axis_joint" type="prismatic">
    <origin rpy="0 0 3.14159" xyz="-1.5 0 0.487"/>
    <parent link="linear_axis"/>
    <child link="KR210_2_socket"/>
    <axis xyz="1.0 0.0 0.0"/>
    <limit effort="100" lower="-1.5" upper="-9" velocity="100"/>
  </joint>

  <link name="KR210_2_socket">
    <visual>
      <origin rpy="0 0 0" xyz="-0.059841 0.000000 0.325856"/>
      <geometry>
        <box size="1.139659 1.019947 0.651711"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.059841 0.000000 0.325856"/>
      <geometry>
        <box size="1.139659 1.019947 0.651711"/>
      </geometry>
    </collision>
  </link>

  <joint name="kr210_2_joint_1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.5256"/>
    <parent link="KR210_2_socket"/>
    <child link="KR210_2_A1"/>
    <axis xyz="0.0 0.0 1.0"/>
    <limit effort="100" lower="-3.22886" upper="3.22886" velocity="100"/>
  </joint>

  <link name="KR210_2_A1">
    <visual>
      <origin rpy="0 0 0" xyz="-0.014046 -0.009306 0.336047"/>
      <geometry>
        <box size="0.631088 1.113008 0.692095"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.014046 -0.009306 0.336047"/>
      <geometry>
        <box size="0.631088 1.113008 0.692095"/>
      </geometry>
    </collision>
  </link>

  <joint name="kr210_2_joint_2" type="revolute">
    <origin rpy="0 0 0" xyz="0.005 -0.35 0.45"/>
    <parent link="KR210_2_A1"/>
    <child link="KR210_2_A2"/>
    <axis xyz="1.0 0.0 0.0"/>
    <limit effort="10000" lower="-2.0944" upper="1.22173" velocity="100"/>
  </joint>

  <link name="KR210_2_A2">
    <visual>
      <origin rpy="0 0 0" xyz="-0.219836 -0.002396 0.656578"/>
      <geometry>
        <box size="0.449671 0.532510 1.714843"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="-0.219836 -0.002396 0.656578"/>
      <geometry>
        <box size="0.449671 0.532510 1.714843"/>
      </geometry>
    </collision>
  </link>

  <joint name="kr210_2_joint_3" type="revolute">
    <origin rpy="0 0 0" xyz="-0.183 0 1.35"/>
    <parent link="KR210_2_A2"/>
    <child link="KR210_2_A3"/>
    <axis xyz="1.0 0.0 0.0"/>
    <limit effort="100" lower="-2.0944" upper="2.70526" velocity="100"/>
  </joint>

  <link name="KR210_2_A3">
    <visual>
      <origin rpy="0 0 0" xyz="0.263750 -0.262250 -0.034810"/>
      <geometry>
        <box size="0.527500 1.399500 0.429352"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.263750 -0.262250 -0.034810"/>
      <geometry>
        <box size="0.527500 1.399500 0.429352"/>
      </geometry>
    </collision>
  </link>

  <joint name="kr210_2_joint_4" type="revolute">
    <origin rpy="0 0 0" xyz="0.178 -0.962 -0.041"/>
    <parent link="KR210_2_A3"/>
    <child link="KR210_2_A4"/>
    <axis xyz="0.0 1.0 0.0"/>
    <limit effort="100" lower="-6.10865" upper="6.10865" velocity="100"/>
  </joint>

  <link name="KR210_2_A4">
    <visual>
      <origin rpy="0 0 0" xyz="0.000000 -0.258565 -0.000750"/>
      <geometry>
        <box size="0.246000 0.537131 0.247500"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.000000 -0.258565 -0.000750"/>
      <geometry>
        <box size="0.246000 0.537131 0.247500"/>
      </geometry>
    </collision>
  </link>

  <joint name="kr210_2_joint_5" type="revolute">
    <origin rpy="0 0 0" xyz="0 -0.438 0"/>
    <parent link="KR210_2_A4"/>
    <child link="KR210_2_A5"/>
    <axis xyz="1.0 0.0 0.0"/>
    <limit effort="100" lower="-2.18166" upper="2.18166" velocity="100"/>
  </joint>

  <link name="KR210_2_A5">
    <visual>
      <origin rpy="0 0 0" xyz="0.023766 -0.050459 0.000000"/>
      <geometry>
        <box size="0.324961 0.341083 0.240222"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.023766 -0.050459 0.000000"/>
      <geometry>
        <box size="0.324961 0.341083 0.240222"/>
      </geometry>
    </collision>
  </link>

  <joint name="kr210_2_joint_6" type="revolute">
    <origin rpy="0 0 0" xyz="0 -0.24 0"/>
    <parent link="KR210_2_A5"/>
    <child link="KR210_2_A6"/>
    <axis xyz="0.0 1.0 0.0"/>
    <limit effort="100" lower="-6.10865" upper="6.10865" velocity="2"/>
  </joint>

  <link name="KR210_2_A6">
    <visual>
      <origin rpy="0 0 0" xyz="0.000000 0.017500 0.000000"/>
      <geometry>
        <box size="0.2 0.035 0.2"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.000000 0.017500 0.000000"/>
      <geometry>
        <box size="0.2 0.035 0.2"/>
      </geometry>
    </collision>
  </link>

  <!-- <link name="EndEffector_Link" />

  <joint name="EndEffector" type="fixed">
    <origin xyz="0 0 -0.061525" rpy="0 0 0" />
    <parent link="KR210_2_A6" />
    <child link="EndEffector_Link" />
    <axis xyz="0 0 0" />
  </joint> -->

</robot>
